pkg/liqoctl/status/peer/peerinfo.go

```diff
@@ -0,0 +1,340 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statuspeer
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/maps"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	liqoconsts "github.com/liqotech/liqo/pkg/consts"
+	"github.com/liqotech/liqo/pkg/liqoctl/output"
+	"github.com/liqotech/liqo/pkg/liqoctl/status"
+	"github.com/liqotech/liqo/pkg/liqoctl/status/utils/resources"
+	liqonetutils "github.com/liqotech/liqo/pkg/liqonet/utils"
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+	foreigncluster "github.com/liqotech/liqo/pkg/utils/foreignCluster"
+	liqogetters "github.com/liqotech/liqo/pkg/utils/getters"
+	liqolabels "github.com/liqotech/liqo/pkg/utils/labels"
+	peeringconditionsutils "github.com/liqotech/liqo/pkg/utils/peeringConditions"
+)
+
+// PeerInfoChecker implements the Check interface.
+// holds the information about the peered cluster.
+type PeerInfoChecker struct {
+	options            *status.Options
+	peerInfoSection    output.Section
+	collectionErrors   []error
+	notFound           bool
+	RemoteClusterNames []string
+}
+
+// NewPeerInfoChecker return a new PeerInfoChecker.
+func NewPeerInfoChecker(o *status.Options, remoteClusterNames ...string) *PeerInfoChecker {
+	return &PeerInfoChecker{
+		peerInfoSection:    output.NewRootSection(),
+		options:            o,
+		RemoteClusterNames: remoteClusterNames,
+		notFound:           false,
+	}
+}
+
+// Silent implements the Checker interface.
+func (pic *PeerInfoChecker) Silent() bool {
+	return false
+}
+
+// Collect implements the collect method of the Checker interface.
+// it collects the infos of the peered cluster.
+func (pic *PeerInfoChecker) Collect(ctx context.Context) {
+	localClusterIdentity, err := liqoutils.GetClusterIdentityWithControllerClient(ctx, pic.options.CRClient, pic.options.LiqoNamespace)
+	localClusterName := localClusterIdentity.ClusterName
+	if err != nil {
+		pic.addCollectionError(fmt.Errorf("unable to get local cluster identity: %w", err))
+		return
+	}
+
+	foreignClusterMap, err := liqogetters.MapForeignClustersByLabel(ctx, pic.options.CRClient, labels.Everything())
+	if err != nil {
+		pic.addCollectionError(fmt.Errorf("unable to get foreign clusters: %w", err))
+		return
+	}
+
+	foreignClusterListSelected := pic.getForeignClusterListSelected(foreignClusterMap)
+
+	if len(foreignClusterListSelected.Items) == 0 {
+		pic.peerInfoSection.AddEntry(fmt.Sprintf("Local cluster %q", localClusterName), "is not peered with any remote cluster")
```

Comment:
The output here seems a bit strange to me, due to the change in color (in part white, and in part orange), and the colon in the middle of the sentence.

```
Local cluster "red-water": is not peered with any remote cluster 
```